internal/cli/enable.go

```diff
@@ -0,0 +1,174 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/tilt-dev/tilt/internal/analytics"
+	engineanalytics "github.com/tilt-dev/tilt/internal/engine/analytics"
+	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
+	"github.com/tilt-dev/tilt/pkg/model"
+)
+
+type enableCmd struct {
+	all  bool
+	only bool
+}
+
+func newEnableCmd() *enableCmd {
+	return &enableCmd{}
+}
+
+func (c *enableCmd) name() model.TiltSubcommand { return "enable" }
+
+func (c *enableCmd) register() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                   "enable {-all | [--only] <resource>...}",
+		DisableFlagsInUseLine: true,
+		Short:                 "Enables resources",
+		Long: `Enables the specified resources in Tilt.
+
+# enables the resources named 'frontend' and 'backend'
+tilt enable frontend backend
+
+# enables frontend and backend and disables all others
+tilt enable --only frontend backend
+
+# enables all resources
+tilt enable --all
+`,
+	}
+
+	addConnectServerFlags(cmd)
+	cmd.Flags().BoolVar(&c.only, "only", false, "Enable the specified resources, disable all others")
+	cmd.Flags().BoolVar(&c.all, "all", false, "Enable all resources")
+
+	return cmd
+}
+
+func (c *enableCmd) run(ctx context.Context, args []string) error {
+	ctrlclient, err := newClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	if c.all {
+		if c.only {
+			return errors.New("cannot use --all with --only")
+		} else if len(args) > 0 {
+			return errors.New("cannot use --all with resource names")
+		}
+	} else if len(args) == 0 {
+		return errors.New("must specify at least one resource")
+	}
+
+	a := analytics.Get(ctx)
+	cmdTags := engineanalytics.CmdTags(map[string]string{})
+	cmdTags["only"] = strconv.FormatBool(c.only)
+	cmdTags["all"] = strconv.FormatBool(c.all)
+	a.Incr("cmd.enable", cmdTags.AsMap())
+	defer a.Flush(time.Second)
+
+	names := make(map[string]bool)
+	for _, name := range args {
+		names[name] = true
+	}
+
+	unselectedState := v1alpha1.DisableStatePending
+	if c.only {
+		unselectedState = v1alpha1.DisableStateDisabled
+	} else if c.all {
+		unselectedState = v1alpha1.DisableStateEnabled
+	}
+	err = changeEnabledResources(ctx, ctrlclient, args, v1alpha1.DisableStateEnabled, unselectedState)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Changes which resources are enabled in Tilt.
+// For resources in `selectedResources`, ensures their disabled state is `selectedState`.
+// For all other resources:
+// - if `unselectedState` is Enabled or Disabled, ensure their disabled state is that
+// - otherwise, ignore them
+func changeEnabledResources(
+	ctx context.Context,
+	cli client.Client,
+	selectedResources []string,
+	selectedState v1alpha1.DisableState,
+	unselectedState v1alpha1.DisableState,
+) error {
```

Comment:
```suggestion
	unselectedState v1alpha1.DisableState) error {
```
this seems easier to read

Reply:
really? that leaves us without anything visually breaking up the params and the code:
![image](https://user-images.githubusercontent.com/7453991/152569696-e7b79ddd-bcf6-4a05-98e4-f3f9eb40be67.png)